cmake-modules/AddTestCMocka.cmake

```diff
@@ -74,12 +74,12 @@ function(ADD_CMOCKA_TEST _TARGET_NAME)
     if (DEFINED _add_cmocka_test_LINK_TARGETS)
         # link to user defined
         target_link_libraries(${_TARGET_NAME}
-            PRIVATE ${CMOCKA_LIBRARIES} ${_add_cmocka_test_LINK_TARGETS} ${MATH_LIB_UNIX}
+            PRIVATE cmocka ${_add_cmocka_test_LINK_TARGETS} ${MATH_LIB_UNIX}
```

Comment:
this is problematic for CI and any user using VCPKG to install cmocka because `cmocka` won't be found like that.

Can you try it like this:
```cmake
    # set cmocka lib include path depending on user working with VCPKG or having cmocka installed globally
    if(DEFINED ENV{VCPKG_ROOT} OR DEFINED ENV{VCPKG_INSTALLATION_ROOT})
        set(CMOCKA_LIB ${CMOCKA_LIBRARIES})
    else()
        set(CMOCKA_LIB cmocka)
    endif()

    if (DEFINED _add_cmocka_test_LINK_TARGETS)
        # link to user defined
        target_link_libraries(${_TARGET_NAME}
            PRIVATE ${CMOCKA_LIB} ${_add_cmocka_test_LINK_TARGETS} ${MATH_LIB_UNIX}
        )   
    else()
        # link against az_core by default
        target_link_libraries(${_TARGET_NAME}
            PRIVATE ${CMOCKA_LIB} az_core ${MATH_LIB_UNIX}
        )
    endif()
```

Reply:
This should make CI pass and also your local dev,
I am not sure about cmocka headers are going to be added as well for you. But, let me know if doing this works for you.
Please, Let me know if you want me to create a PR for this or if you can update this PR and I can approve and merge

Reply:
Absolute, makes sense. Fixed as suggested.